api/v1alpha1/oauth2client_types.go

```diff
@@ -69,6 +72,10 @@ type GrantType string
 // ResponseType represents an OAuth 2.0 response type strings
 type ResponseType string
 
+// +kubebuilder:validation:Pattern=^https?://.*
+// RedirectURI represents a redirect URI for the client
```

Comment:
Nit: Oauth2 allows redirect_uri to be a string URI with custom scheme which are used for example in moblie native applications.

Reply:
Fixed in 3133f9a7c633596e9319c815c33a529de55ab90b

Reply:
TBH this regex will not fully work either, it will match for example `http://http://whatever.com` `http://`  `http:/`
Validating it in the code manually using `/net/url` package is also not working properly as well. I mean url.Parse && url.ParseRequestURI. Let's leave it for now as it is